src/test/moves/fairy_lock.test.ts

```diff
@@ -0,0 +1,149 @@
+import { ArenaTagSide } from "#app/data/arena-tag";
+import { ArenaTagType } from "#app/enums/arena-tag-type";
+import { Abilities } from "#enums/abilities";
+import { Moves } from "#enums/moves";
+import { Species } from "#enums/species";
+import GameManager from "#test/utils/gameManager";
+import Phaser from "phaser";
+import { afterEach, beforeAll, beforeEach, describe, expect, it } from "vitest";
+
+describe("Moves - Fairy Lock", () => {
+  let phaserGame: Phaser.Game;
+  let game: GameManager;
+
+  beforeAll(() => {
+    phaserGame = new Phaser.Game({
+      type: Phaser.HEADLESS,
+    });
+  });
+
+  afterEach(() => {
+    game.phaseInterceptor.restoreOg();
+  });
+
+  beforeEach(() => {
+    game = new GameManager(phaserGame);
+    game.override
+      .moveset([ Moves.FAIRY_LOCK, Moves.SPLASH ])
+      .ability(Abilities.BALL_FETCH)
+      .battleType("double")
+      .disableCrits()
+      .enemySpecies(Species.MAGIKARP)
+      .enemyAbility(Abilities.BALL_FETCH)
+      .enemyMoveset([ Moves.SPLASH, Moves.U_TURN ]);
+  });
+
+  it("Applies Fairy Lock tag for one turn, then apply Trapped tag for one turn", async () => {
+    await game.classicMode.startBattle([ Species.KLEFKI, Species.TYRUNT ]);
+    const playerPokemon = game.scene.getPlayerField();
+    const enemyField = game.scene.getEnemyField();
+
+    game.move.select(Moves.FAIRY_LOCK);
+    game.move.select(Moves.SPLASH, 1);
+    await game.forceEnemyMove(Moves.SPLASH, 1);
+    await game.forceEnemyMove(Moves.SPLASH, 1);
+    await game.phaseInterceptor.to("BerryPhase");
+    expect(game.scene.arena.getTagOnSide(ArenaTagType.FAIRY_LOCK, ArenaTagSide.PLAYER)).toBeDefined();
+    expect(game.scene.arena.getTagOnSide(ArenaTagType.FAIRY_LOCK, ArenaTagSide.ENEMY)).toBeDefined();
+
+    await game.toNextTurn();
+
+    game.move.select(Moves.SPLASH);
+    game.move.select(Moves.SPLASH);
+    await game.forceEnemyMove(Moves.SPLASH, 1);
+    await game.forceEnemyMove(Moves.SPLASH, 1);
+    await game.phaseInterceptor.to("BerryPhase");
+    expect(playerPokemon[0].isTrapped()).toEqual(true);
+    expect(playerPokemon[1].isTrapped()).toEqual(true);
+    expect(enemyField[0].isTrapped()).toEqual(true);
+    expect(enemyField[1].isTrapped()).toEqual(true);
+
+    await game.toNextTurn();
+    expect(playerPokemon[0].isTrapped()).toEqual(false);
+    expect(playerPokemon[1].isTrapped()).toEqual(false);
+    expect(enemyField[0].isTrapped()).toEqual(false);
+    expect(enemyField[1].isTrapped()).toEqual(false);
+  });
```

Comment:
```suggestion
  });

```
Give the tests some breathing room lol